### PR TITLE
Harris corner detection

### PR DESCRIPTION
**Pull request by jeffin07**
### **Describe your change:**

Added Harris corner detection. #1809 

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: computer_vision/harriscorner.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+"""
+Harris Corner Detector
+https://en.wikipedia.org/wiki/Harris_Corner_Detector
+"""
+
+
+class Harris_Corner:
+    def __init__(self, k, window_size):
+
+        if k in [0.04, 0.06]:
```
Comment (cclauss) [inline computer_vision/harriscorner.py:13]:
```suggestion
        if k in (0.04, 0.06):
```
A tuple takes slightly less memory than a list and signals that its contents can not be modified at runtime.

File: computer_vision/harriscorner.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            self.k = k
+            self.window_size = window_size
+        else:
+            print("invalid k value")
```
Comment (cclauss) [inline computer_vision/harriscorner.py:17]: More Pythonic to `raise ValueError("Invalid k value")`

File: computer_vision/harriscorner.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import cv2
+
+"""
+Harris Corner Detector
+https://en.wikipedia.org/wiki/Harris_Corner_Detector
+"""
+
+
+class Harris_Corner:
+    def __init__(self, k, window_size):
```
Comment (cclauss) [inline computer_vision/harriscorner.py:11]: Type hints?  What is `k`?  Could we have a more self documenting variable name?

File: computer_vision/harriscorner.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+        if k in [0.04, 0.06]:
+            self.k = k
+            self.window_size = window_size
+        else:
+            print("invalid k value")
+            exit()
+
+    def __str__(self):
+        return "Harris Corner  detection with k : {}".format(self.k)
```
Comment (cclauss) [inline computer_vision/harriscorner.py:21]:
```suggestion
        return f"Harris Corner detection with k: {self.k}"
```

File: computer_vision/harriscorner.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+    def detect(self, img_path):
```
Comment (cclauss) [inline computer_vision/harriscorner.py:23]: Type hints?

File: computer_vision/harriscorner.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        """
+        Returns the image with corners identified
+        img_path  : path of the image
+        output : list of the corner positions, image
+        """
+        img = cv2.imread(img_path, 0)
+        h, w = img.shape
+        corner_list = []
+        color_img = img.copy()
+        color_img = cv2.cvtColor(color_img, cv2.COLOR_GRAY2RGB)
```
Comment (TheSuperNoob) [inline computer_vision/harriscorner.py:33]: Is this converting the image from grayscale to RGB or is it turning RGB to grayscale? The wikipedia page says to turn it to grayscale.

Comment (jeffin07) [reply computer_vision/harriscorner.py:40]: Actually that's for the output image.

File: computer_vision/harriscorner.py (hunk 3 above)
Comment (jeffin07) [reply computer_vision/harriscorner.py]: it's a constant, also represented the same in wiki page

**Comment by cclauss** [comment]
I was a bit too fast in merging this.  Do we ___really___ need a whole new directory for computer vision?  Couldn't this instead fit somewhere in
https://github.com/TheAlgorithms/Python/tree/master/digital_image_processing ?

**Comment by jeffin07** [comment]
I think Image processing and computer vision and little different  from each other.Image processing means applying some algorithms for transforming image from one form to other like smoothing,contrasting, stretching etc While in computer vision  comes from modelling image processing using the techniques of machine learning.Computer vision applies machine learning to recognize patterns for interpretation of images. Much like the process of visual reasoning of human vision

**Comment by poyea** [comment]
That's great. Feel free to open an issue, or create a readme inside the folder for description, just like the one in `/bit_manipulation/`.

**Pull request merged**